openedx/core/djangoapps/content/course_overviews/models.py

```diff
@@ -362,6 +421,61 @@ def pre_requisite_courses(self):
         return json.loads(self._pre_requisite_courses_json)
 
     @classmethod
+    def get_select_courses(cls, course_keys):
+        """
+        Returns CourseOverview objects for the given course_keys.
+        """
+        course_overviews = []
+
+        log.info('Generating course overview for %d courses.', len(course_keys))
+        log.debug('Generating course overview(s) for the following courses: %s', course_keys)
+
+        for course_key in course_keys:
+            try:
+                course_overviews.append(CourseOverview.get_from_id(course_key))
+            except Exception as ex:  # pylint: disable=broad-except
+                log.exception(
+                    'An error occurred while generating course overview for %s: %s',
+                    unicode(course_key),
+                    ex.message,
+                )
+
+        log.info('Finished generating course overviews.')
+
+        return course_overviews
+
+    @classmethod
+    def get_all_courses(cls, force_reseeding=False):
+        """
+        Returns all CourseOverview objects in the database.
+
+        Arguments:
+            force_reseeding (bool): Optional parameter.
+                If True, the modulestore is used as the source of truth for
+                the list of courses, even if the CourseOverview table was
+                previously seeded.
+
+                If False, the list of courses is retrieved from the
+                CourseOverview table if it was previously seeded, falling
+                back to the modulestore if it wasn't seeded.
+        """
+        if force_reseeding or not CourseOverviewGeneratedHistory.objects.first():
+            # Seed the CourseOverview table with data for all
+            # courses in the system.
+            course_keys = [course.id for course in modulestore().get_courses()]
+            course_overviews = cls.get_select_courses(course_keys)
+            CourseOverviewGeneratedHistory.objects.create(num_courses=len(course_overviews))
+
+        else:
+            # Note: If a newly created course is not returned in this QueryList,
+            # make sure the "publish" signal was emitted when the course was
+            # created.  For tests using CourseFactory, use emit_signals=True.
+            # Or pass True for force_reseeding.
+            course_overviews = CourseOverview.objects.all()
```

Comment:
I'm not a designated reviewer on this PR. However, could this be further improved by performing filtering as part of this query, as opposed to later on in `branding.get_visible_courses()`? The previous implementation appears to have attempted this by passing `filtered_by_org` to the modulestore's `get_courses()`. Just a thought.